lib/pg_party/model/shared_methods.rb

```diff
@@ -8,7 +8,10 @@ module SharedMethods
       def reset_primary_key
         if self != base_class
           base_class.primary_key
-        elsif partition_name = partitions.first
+        elsif (partitions = partitions(include_subpartitions: true)) && partitions.any?
```

Comment:
I was never super thrilled that I was assigning a variable within this `elsif` statement. Now that it's a bit more complex maybe we should consider refactoring this method

Reply:
Refactored this method in https://github.com/rkrage/pg_party/pull/54/commits/b1783ffe1b70217480c818fd67701a191ff90264